pkg/lib/serverTime.js

```diff
@@ -321,13 +324,181 @@ export function ServerTime() {
+        return cockpit.file(names[first], { superuser: "try" }).read()
+                .then(text => {
+                    if (text)
+                        return [names[first], text];
+                    return find_file_and_read(names, first + 1);
+                });
+    }
+
+    function get_ntp_config_chronyd() {
+        return find_file_and_read(["/etc/chrony.conf", "/etc/chrony/chrony.conf"])
+                .then(([name, text]) => {
+                    const result = {
+                        supported: true,
+                        backend: "chronyd",
+                        file: name,
+                        standard: [],
+                        sources: [],
+                        extra: []
+                    };
+
+                    let in_cockpit_section = false;
+                    for (const line of text.split("\n")) {
+                        if (line == "# Added by Cockpit") {
+                            in_cockpit_section = true;
+                        } else {
+                            const fields = line.split(" ");
+                            if (fields[0] == "server") {
+                                if (in_cockpit_section) {
+                                    result.extra.push(fields[1]);
+                                } else {
+                                    result.standard.push({ name: fields[1], enabled: true });
+                                }
+                            } else {
+                                in_cockpit_section = false;
+                                if (fields[0] == "pool")
+                                    result.standard.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#pool" || fields[0] == "#server")
+                                    result.standard.push({ name: fields[1], enabled: false });
+                                if (fields[0] == "sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: false });
+                            }
+                        }
+                    }
+
+                    return result;
+                });
+    }
+
+    function set_ntp_config_chronyd(config) {
+        const chrony_file = cockpit.file(config.file, { superuser: "try" });
+
+        function confline(line, list, name) {
+            const c = list.find(e => e.name == name);
+            if (!c)
+                return line;
+            const line_trim = line.trim();
+            const line_enabled = !line_trim.startsWith("#");
+            if (c.enabled && !line_enabled)
+                return line_trim.substr(1);
+            if (!c.enabled && line_enabled)
+                return "#" + line_trim;
+            return line;
+        }
+
+        return chrony_file.modify(text => {
```

Comment:
I'm not commenting on this yet, as it's tightly related to the design questions raised above. Let's sort these out first.